IPython/core/ultratb.py

```diff
@@ -394,20 +394,16 @@ def _format_traceback_lines(lnum, index, lines, Colors, lvals=None, scheme=None)
     res = []
     i = lnum - index
 
-    # This lets us get fully syntax-highlighted tracebacks.
-    if scheme is None:
-        ipinst = get_ipython()
-        if ipinst is not None:
-            scheme = ipinst.colors
-        else:
-            scheme = DEFAULT_SCHEME
-
+    global _parser
+    if scheme and (_parser.style != scheme):
+        _parser = _parser = PyColorize.Parser(style=scheme)
```

Comment:
Does this actually need to be a global? Instantiating a `Parser` looks like it's cheap, so we could just do it each time we need it.

Also, you're assigning to the same name twice here.